### ISSUE TITLE
Create Training loop function

### ISSUE THREAD

**Issue by dannu444** [dannu444/4al3-IoT-threatdetector#6]
- Implement function to execute stochastic gradient descent with L2 regularization and early stopping
- Should take as input the model, train/test splits for X and y, patience, how often to save train/test loss and accuracy, total iterations, batch size, and any other inputs required by pytorch
- Should output statistics (final train/val loss and accuracy) as well as data for plots of loss/accuracy
